h2o-algos/src/main/java/hex/tree/SharedTreeModel.java

```diff
@@ -175,6 +175,39 @@ public SharedTreeOutput( SharedTree b) {
       _modelClassDist = _priorClassDist;
     }
 
+    @Override
+    public TwoDimTable createInputFramesInformationTable(ModelBuilder modelBuilder) {
+      List<String> colHeaders = new ArrayList<>();
+      List<String> colTypes = new ArrayList<>();
+      List<String> colFormat = new ArrayList<>();
+      SharedTreeParameters params = (SharedTreeParameters) modelBuilder._parms;
+
+      colHeaders.add("Input Frame"); colTypes.add("string"); colFormat.add("%s");
+      colHeaders.add("Checksum"); colTypes.add("long"); colFormat.add("%d");
+      colHeaders.add("ESPC"); colTypes.add("string"); colFormat.add("%d");
+
+      final int rows = 3;
+      TwoDimTable table = new TwoDimTable(
+              "Input Frames Information", null,
+              new String[rows],
+              colHeaders.toArray(new String[0]),
+              colTypes.toArray(new String[0]),
+              colFormat.toArray(new String[0]),
+              "");
+
+      table.set(0, 0, "training_frame");
+      table.set(1, 0, "validation_frame");
+      table.set(2, 0, "calibration_frame");
+      table.set(0, 1, modelBuilder.train() != null ? modelBuilder.train().checksum() : -1);
+      table.set(1, 1, params._valid != null ? modelBuilder.valid().checksum() : -1);
```

Comment:
should this be modelBuilder.valid() != null? you can set the train/valid Frames directly